### ISSUE TITLE
questions about ECAPA-TDNN model training

### ISSUE THREAD

**Issue by forwiat** [lawlict/ECAPA-TDNN#1]
hello lawlict, Thank you so much for the great work you shared.
My eer is 2.65% in voxceleb_test which can't reach to 1.*% when I training this model. I use aam-softmax loss which is same as loss mentioned in paper. My batch set to 512 and lr init = 0.001, total steps = 240000 and reduce lr in [120000, 180000]. I wonder your experiment details in training. my experiment setting must significantly be same as paper's parameters?

**Comment by lawlict** [comment]
Hi @forwiat. The outstanding performance of ECAPA-TDNN doesn't fully rely on the model structure. In fact, there are lots of details that help improve the performance. For example:

- 6 kinds of data augmentation
- a cyclical learning rate
- adaptive s-norm

I recommend that you use this model based on voxceleb_trainer(https://github.com/clovaai/voxceleb_trainer). Still, I only achieved a EER of 1.2% on the original voxceleb1_test set. Therefore I guess that some details are still missing.

**Comment by forwiat** [comment]
OK, I will carefully try it again!
Thanks for your quick reply and answer!

**Comment by TaoRuijie** [comment]
Hi, thanks for your sharing!

So you are using the same code in voxceleb_trainer, instead of changing the model into this ecapa-tdnn right? And you can get 1.2 EER in vox1 test set. Is my understanding correct?

**Comment by lawlict** [comment]
Hi, @TaoRuijie. Yes, your understanding is correct.

**Comment by seacj** [comment]
Hi, I also train the ECAPA-TDNN model (C=1024) with voxceleb_trainer but get 1.19 EER . After using as-norm, I get 1.08 EER. 
Do you have any suggestion to get 0.87 EER in the paper?

**Comment by lawlict** [comment]
> Hi @forwiat. The outstanding performance of ECAPA-TDNN doesn't fully rely on the model structure. In fact, there are lots of details that help improve the performance. For example:
> 
> * 6 kinds of data augmentation
> * a cyclical learning rate
> * adaptive s-norm
> 
> I recommend that you use this model based on voxceleb_trainer(https://github.com/clovaai/voxceleb_trainer). Still, I only achieved a EER of 1.2% on the original voxceleb1_test set. Therefore I guess that some details are still missing.

Hi, @seacj. As you see, I only achieve a EER of 1.2% on Voxceleb-1-O.trials without as-norm. I guess that the original paper uses Voxceleb1-O-clean.trials, which is slightly easier.
